src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -114,60 +145,123 @@ export const useBalanceTableColumns = (
-      items: [
-        {
-          key: 'add_funds',
-          onClick: () => {},
-          label: '',
-          icon: '',
-        },
-        // {
-        //   key: 'view_ethscan',
-        //   onClick: () => {},
-        //   icon: 'arrow-square-out',
-        //   label: formatText(
-        //     { id: 'membersPage.memberNav.viewOn' },
-        //     {
-        //       networkName: DEFAULT_NETWORK_INFO.blockExplorerName,
-        //     },
-        //   ),
-        //   renderItemWrapper: (props, children) => (
-        //     <Link
-        //       to={getBlockExplorerLink({
-        //         linkType: 'address',
-        //         addressOrHash: walletAddress,
-        //       })}
-        //       {...props}
-        //     >
-        //       {children}
-        //     </Link>
-        //   ),
-        // },
-        // {
-        //   key: 'mint_tokens',
-        //   onClick: () => {},
-        //   label: 'Mint tokens',
-        //   icon: 'bank',
-        // },
-        // {
-        //   key: 'transfer_fundss',
-        //   onClick: () => {},
-        //   label: 'Transfer funds',
-        //   icon: 'transfer',
-        // },
-        // {
-        //   key: 'make_payment',
-        //   onClick: () => {},
-        //   label: 'Make payment with token',
-        //   icon: 'hand-coins',
-        // },
-      ],
-    };
-  }, []);
+  >(
+    ({ index }) => {
+      const selectedTokenData = data[index]?.token;
+      const isTokenNative =
+        selectedTokenData?.tokenAddress === nativeToken?.tokenAddress;
+
+      return {
+        cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
+        items: [
+          ...(!isTokenNative && !nativeTokenStatus?.unlocked
+            ? [
+                {
+                  key: 'add_funds',
+                  onClick: () => {
+                    toggleAddFundsModalOn();
+                  },
+                  label: formatMessage(MSG.labelAddFunds),
+                  icon: 'add',
+                },
+              ]
+            : []),
+          {
+            key: 'view_ethscan',
+            renderItemWrapper: (props, children) => (
+              <Link
+                to={getBlockExplorerLink({
+                  linkType: 'token',
+                  addressOrHash: selectedTokenData?.tokenAddress || '',
+                })}
+                {...props}
+              >
+                {children}
+              </Link>
+            ),
+            label: formatMessage(MSG.labelEthscan),
+            icon: 'arrow-square-out',
+          },
+          ...(isTokenNative
+            ? [
+                {
+                  key: 'mint_tokens',
+                  onClick: () => {
+                    dispatchGlobalEvent(GLOBAL_EVENTS.SET_ACTION_TYPE, {
+                      detail: {
+                        actionType: ACTION.MINT_TOKENS,
+                      },
+                    });
+                    toggleActionSidebarOn();
```

Comment:
I believe the latest way to do this is to pass the initial values to the `toggleActionSidebarOn` function directly, like here:
<img width="449" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/112586815/8232bc30-0c53-47ff-9044-4ae384ea19f6">